sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorReceiver.java

```diff
@@ -405,6 +447,7 @@ private Mono<Boolean> beginClose(ErrorCondition errorCondition) {
                 sink.success(localCloseScheduled);
             }
         });
+        return handler.beginClose(localCloseMono);
```

Comment:
I think this is odd because if it returns a mono, we should just chain it with the next operator. I looked at the current implementation and could we do something like this instead to make it more reactive?
```java
   Mono<Boolean> beginClose() {
        if (isV2) {
            return receiveLinkHandler2.preClose();
        } else {
            return Mono.empty();
        }
    }
```
calling code:
```
handler.beginClose().then(localCloseMono);
```

Reply:
Yes, this is better, let’s follow this pattern, thanks!